### PR TITLE
Create a IANA types file

### PR DESCRIPTION
**Pull request by mjethanandani**
The IANA types file will capture all the feature and identity definitions in one place.

Capturing it in a IANA file allows easier upgrades to deal with updates of the model.

The changes are HUGE, but they are mostly semantic. I would suggest that if you are planning to make changes, wait for this change to get merged. Otherwise, you will be dealing with a lot of conflicts.

Should address issue #26, issue #27, issue #28.

### CODE REVIEW

File: src/yang/iana-qos-types.yang [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+    description
+      "Initial version";
+    reference
+      "RFC XXXX: YANG Models for Quality of Service (QoS).";
+  }
+
+  /*
+   * Features.
+   */
+  feature meter-template-support {
```
Comment (aseemavi) [inline src/yang/iana-qos-types.yang:44]: you may want to update the codebase. Most of these features were removed.

**Review by aseemavi — COMMENTED** [review]
some changes to ietf-qos-oper.yang too

File: src/yang/iana-qos-types.yang (hunk 1 above)
Comment (mjethanandani) [reply src/yang/iana-qos-types.yang:44]: @aseemavi , I am up-to-date with whatever is in the codebase. If you removed any features, they must be local to you.

Comment (aseemavi) [reply src/yang/iana-qos-types.yang:44]: In latest published draft (08), there are only 5 features:
child-policy-feature
count-feature
named-counter-feature
policy-inline-classifier-config
classifier-template-feature

Rest were cleaned up. If it is part of code base and not part of latest published draft, then we have to see how it happened.

Comment (mjethanandani) [reply src/yang/iana-qos-types.yang]: Ok. I have removed all but the five above features from all the files. Also there are two grouping that were not being referenced by anyone. Do not know if they got missed or are no longer needed. 

Please review the changes closely to make sure we have not missed anything.

**Comment by mjethanandani** [comment]
> some changes to ietf-qos-oper.yang too

Not sure exactly what the changes are. Can you expand?

**Comment by aseemavi** [comment]
> > some changes to ietf-qos-oper.yang too
> 
> Not sure exactly what the changes are. Can you expand?

These identities can be moved to yang-types as well, if already not done:

  identity direction {
    description
      "This is a base identity of traffic direction";
  }
  identity inbound {
    base direction;
    description
      "Direction of traffic coming into the network entry";
  }
  identity outbound {
    base direction;
    description
      "Direction of traffic going out of the network entry";
  }

**Comment by mjethanandani** [comment]
> > > some changes to ietf-qos-oper.yang too
> > 
> > 
> > Not sure exactly what the changes are. Can you expand?
> 
> These identities can be moved to yang-types as well, if already not done:
> 
> identity direction { description "This is a base identity of traffic direction"; } identity inbound { base direction; description "Direction of traffic coming into the network entry"; } identity outbound { base direction; description "Direction of traffic going out of the network entry"; }

They have already been moved (and removed from the oper file.

**Review by aseemavi — APPROVED** [review]

**Pull request merged**